compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1100,7 +1101,11 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
                     let vis = self.get_visibility(child_index);
                     let def_id = self.local_def_id(child_index);
                     let res = Res::Def(kind, def_id);
-                    callback(Export { res, ident, vis, span });
+
+                    if !kind.is_macro() {
+                        callback(Export { res, ident, vis, span });
+                    }
```

Comment:
Could you add a comment/FIXME explaining that macros are currently encoded twice as items and as reexports, so we are ignoring the items when decoding module children and only use the reexports.

Reply:
Is there a reasonably simple way to fix it? This workaround feels very hacky to me.

Reply:
Yes, something like https://github.com/cjgillot/rust/commit/c147d8fdf8c4428ac94216f9629816b357c05435.
I think it's ok to land this as is in this PR.

Reply:
Alright, I'll leave it this way for now and do the comment.